llvm/test/Transforms/LoopVectorize/single-early-exit-deref-assumptions.ll

```diff
@@ -94,3 +94,50 @@ loop.end:
   %retval = phi i64 [ %index, %loop ], [ -1, %loop.inc ]
   ret i64 %retval
 }
+
+define i64 @early_exit_alignment_and_deref_known_via_assumption(ptr noalias %p1, ptr noalias %p2, i64 %n) nofree nosync {
+; CHECK-LABEL: define i64 @early_exit_alignment_and_deref_known_via_assumption(
```

Comment:
Hmm, this loop isn't vectorising now that we have the deferenceable assumption so I'm not quite sure what this is actually testing? Is there more work required elsewhere or is it just because the test itself inhibits vectorisation?

Reply:
Looks like we bail out here:

```
  // Try to get the access size.
  const SCEV *PtrDiff = SE.getMinusSCEV(AccessEnd, AccessStart);
  if (isa<SCEVCouldNotCompute>(PtrDiff))
    return false;
```

in isDereferenceableAndAlignedInLoop

Reply:
Looks like AccessEnd = `inttoptr (i64 -1 to ptr)` and AccessStart = `%p1`

Reply:
Hi @fhahn, just letting you know I'm working on a patch to fix this downstream. Hopefully have a PR ready soon!

Reply:
As I mentioned in the earlier conversation, I have been working on a patch, will share soon

Reply:
No worries, just wanted to avoid working on the same thing unnecessarily :)